src/state_point.cpp

```diff
@@ -47,6 +49,15 @@ extern "C" int openmc_statepoint_write(const char* filename, bool* write_source)
       simulation::current_batch, w);
   }
 
+  // If a file name was specified, ensure it has .h5 file extension
+  const auto extension = get_file_extension(filename_);
+  if (extension == "") {
+    filename_.append(".h5");
+  } else if (extension != "h5") {
+    fatal_error("openmc_statepoint_write was passed an incorrect file "
+                "extension. Must either have no file extension or .h5");
```

Comment:
Does it really matter what the extension is? If someone wants to write a statepoint called "gavin.is.awesome", is there a good reason to stop them?

Reply:
I agree, I've changed it to a helpful warning message instead saying that an hdf5 file will be written regardless of the extension passed in.